index.js

```diff
@@ -219,6 +223,110 @@ let ContainerService = {
     });
   },
 
+  getTabs(args) {
+    return new Promise((resolve, reject) => {
+      if (!("userContextId" in args)) {
+        reject(null);
+        return;
+      }
+
+      let list = [];
+      let promises = [];
+
+      for (let tab of tabs) {
+        let xulTab = viewFor(tab);
+        let userContextId = parseInt(xulTab.getAttribute("usercontextid") || 0, 10);
+        if (args.userContextId == userContextId) {
+          let object = { title: tab.title, id: tab.id };
+          list.push(object);
+          promises.push(getFavicon(tab.url).then(url => {
+            object.favicon = url;
+          }, () => {
+            object.favicon = "";
+          }));
+        }
+      }
+
+      Promise.all(promises).then(() => {
+        resolve(list);
+      });
+    });
+  },
+
+  showTab(args) {
+    return new Promise((resolve, reject) => {
+      if (!("tabId" in args)) {
+        reject(null);
+        return;
+      }
+
+      for (let tab of tabs) {
+        if (tab.id == args.tabId) {
+          tab.window.activate();
+          tab.activate();
+          break;
+        }
+      }
+
+      resolve(null);
+    });
+  },
+
+  moveTabsToWindow(args) {
+    return new Promise((resolve, reject) => {
+      if (!("userContextId" in args)) {
+        reject(null);
```

Comment:
Need a reason error message here. e.g., `reject("moveTabsToWindow must be called with userContextId argument.")`